### DISCUSSION TITLE
Question regarding handling of amigaforever kickstart rom files

### DISCUSSION THREAD

**Discussion by giantclambake** [BlitterStudio/amiberry#1078] [General]
Greets... amiberry 5.6.1 x86-64 build ..

Premise: User installs amiberry, and buys AmigaForever v10 (current) to obtain legit kickstart roms ; (linux) user will install the 'msitools' to extract the amigaforever installer.msi file, and copy the required rom files into ~/amiberry/kickstarts , start amiberry, double check path & rescan --- no kickstart roms found....

... of course, checksums don't match those listed in the wiki =)

Any explanation/insights as to what's going on here? (the user's perspective might be 'have I wasted my $10?'  8)

Is there a good workaround?.... ie; add checksum values of AF v10 kickstart roms to amiberry source code somewhere ... or ...do these rom files not work with amiberry, which is hard to test as the GUI doesn't show them (manual config tweek?)

TIA

**Comment by midwan** [comment]
I believe that the Amiga Forever ROMs are usable after you use them once in Amiga Forever, to decrypt them.
They used to come with a `rom.key` file, but that's no longer the case with newer versions. Instead, you just have to launch Amiga Forever / emulation once, and it decrypts the files.

After that, you should be able to copy them over and use them in Amiberry.

**Comment by giantclambake** [comment]
Thanks for the insight ~ msiextract is not enough.

Confirmed - on Debian 11 amd64 ;

*Using WINE, install the amigaforever installer.msi into $wineprefix ..no addons/winetricks needed (I use Play-on-Linux for this)
*Start the AmigaForever.exe installed above with WINE, enter your purchase name/registration number, and click through installer.
*When the AmigaForever GUI appears, pick a configuration ..ie; workbench 3.0 ..and start the emulator (roms decrypted here)

Quit AmigaForever, decrypted roms will be in  $WINEPREFIX/drive_c/users/Public/Documents/Amiga Files/Shared/rom/*.rom

These rom files can be copied into you amiberry/kickstarts directory, and are detected/renamed appropriately in the GUI

Thanks for the solution ~ perhaps guidance to this might  be fodder for the wiki->kickstart rom (bios) page? 

Kudos!

**Reply by midwan** [reply]
Thanks for confirming. I've updated the wiki with a note about this scenario, accordingly.

**Reply by giantclambake** [reply]
Looks good, obviates what needs to be done. 

Closing this as resolved.
